lib/doorkeeper/oauth/base_request.rb

```diff
@@ -28,8 +28,9 @@ def scopes
 
       def find_or_create_access_token(client, resource_owner, scopes, server)
         context = Authorization::Token.build_context(client, grant_type, scopes, resource_owner)
+        application = client.is_a?(Doorkeeper::Application) ? client : client.application if client
```

Comment:
Doing this breaks apps that are using Custom application classes as defined in doorkeeper.rb

`application_class "MyCustomApplicationClass"`

And defined like so
```
# frozen_string_literal: true

class MyCustomApplicationClass < ApplicationRecord
  include ::Doorkeeper::Orm::ActiveRecord::Mixins::Application
end
```